### ISSUE TITLE
Scheduler log messages about Client don't (easily) tell you which client it is.

### ISSUE THREAD

**Issue by rbubley** [dask/distributed#1406]
Currently, the scheduler logs contain information about clients like this:
```
Receive client connection: Client-6f844c0f-9d5d-11e7-8ccd-abcdef10bf84
```

Where the string after "Client" encodes the time the Client class was instantiated and its host's MAC address.

It is tedious to decode this, and doesn't lend itself to easily seeing what is going on.

If we could pass an additional "name" parameter on the instantiation of a Client, this could work its way through to the scheduler, and thus make easier to understand log messages.

**Comment by mrocklin** [comment]
This seems reasonable to me.  We currently keep a `worker_info` dictionary on the scheduler.  I can imagine maintaining a `client_info` dictionary as well.

> Where the string after "Client" encodes the time the Client class was instantiated and its host's MAC address.

Note that this isn't a MAC address.  This is a UUID.

**Comment by rbubley** [comment]
I'll put together a simple PR.

The current client id is based on uuid1, which encodes a few things, including uuid.getnode() which is usually the MAC address:

> uuid.getnode()
Get the hardware address as a 48-bit positive integer. The first time this runs, it may launch a separate program, which could be quite slow. If all attempts to obtain the hardware address fail, we choose a random 48-bit number with its eighth bit set to 1 as recommended in RFC 4122. “Hardware address” means the MAC address of a network interface, and on a machine with multiple network interfaces the MAC address of any one of them may be returned.

**Comment by rbubley** [comment]
Closed via #1415
